atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -284,12 +286,15 @@ private void saveSweepResults(final SweepProgressRowResult progress,
                                   final SweepResults results) {
         final long cellsDeleted = fromNullable(progress.getCellsDeleted()) + results.getCellsDeleted();
         final long cellsExamined = fromNullable(progress.getCellsExamined()) + results.getCellsExamined();
+        final long minimumSweptTimestamp = progress.getMinimumSweptTimestamp() == null ?
+                results.getSweptTimestamp() :
+                Math.min(progress.getMinimumSweptTimestamp(), results.getSweptTimestamp());
```

Comment:
If we set the minimumSweptTimestamp to 0 when null then we should never actually increment the field since progress will not start with a set timestamp field when read from the priority table only when read from the progress table.